src/IECoreMaya/FromMayaMeshConverter.cpp

```diff
@@ -427,6 +445,75 @@ IECoreScene::PrimitiveVariable FromMayaMeshConverter::colors( const MString &col
 	return PrimitiveVariable( PrimitiveVariable::FaceVarying, data );
 }
 
+void FromMayaMeshConverter::corners( std::vector<int> &cornerIds, std::vector<float> &cornerSharpnesses ) const
+{
+	MFnMesh fnMesh( object() );
+
+	MUintArray vertexIds;
+	MDoubleArray creaseData;
+
+	MStatus s = fnMesh.getCreaseVertices( vertexIds, creaseData );
+
+	if( !s )
+	{
+		// Instead of returning empty data in the case where no creases are
+		// present, Maya considers this a failure. We ignore the error status and
+		// return without modifying the given vectors.
+		return;
+	}
+
+	assert( vertexIds.length() == creaseData.length() );
+
+	cornerIds.reserve( vertexIds.length() );
+	cornerSharpnesses.reserve( creaseData.length() );
+
+	for( size_t i = 0; i < vertexIds.length(); ++i )
+	{
+		cornerIds.push_back( vertexIds[i] );
+		cornerSharpnesses.push_back( creaseData[i] );
+	}
+}
+
+void FromMayaMeshConverter::creases( std::vector<int> &creaseLengths, std::vector<int> &creaseIds, std::vector<float> &creaseSharpnesses ) const
+{
+	MFnMesh fnMesh( object() );
+
+	MUintArray edgeIds;
+	MDoubleArray creaseData;
+
+	MStatus s = fnMesh.getCreaseEdges( edgeIds, creaseData );
+	if( !s )
+	{
+		// Instead of returning empty data in the case where no creases are
+		// present, Maya considers this a failure. We ignore the error status and
+		// return without modifying the given vectors.
+		return;
+	}
+
+	assert( edgeIds.length() == creaseData.length() );
+
+	creaseLengths.resize( edgeIds.length(), 2 );
+	creaseIds.reserve( creaseLengths.size() * 2 );
+	creaseSharpnesses.reserve( creaseLengths.size() );
+
+	// Maya stores creases via edge id. Cortex uses vertex ids instead. The
+	// following handles the conversion.
+	// \todo: Cortex supports a more compact crease representation that saves
```

Comment:
is this worth doing now? how much would we save on a realistic production asset?

Reply:
I had a really quick look at a small part of a current asset. We can run a more in-dept analysis if needed. For this particular mesh 260/1516 edges have crease data attached and it's ~20 creased edge loops in total. It looks like the sharpness values are the same within a loop.

I guess that means we'd store 20 floats and ints each instead of 260 for sharpnesses and lengths. And we'd store 280 (every vert stored once except one duplicate per loop) ints instead of 520 for vertex indices. Is that correct?

Assuming we can generalize for the entire asset and for other assets, that sounds like it would give us quite an improvement. Here's a counter argument:

```
>>> print ( o.creaseLengths().memoryUsage() + o.creaseSharpnesses().memoryUsage() + o.creaseIds().memoryUsage() ) / float( o.memoryUsage() )
0.00138504155125
```

Anything wrong with my reasoning here?

Reply:
ok, lets leave it for later if needed